samples/bootloader/README.rst

```diff
@@ -0,0 +1,169 @@
+
+       openssl ecparam -name prime256v1 -genkey -noout -out priv.pem
+
+   OpenSSL is installed with GIT, so it should be available in your GIT bash.
+   See `openSSL`_ for more information.
+
+   .. note::
+      This step is optional for testing the bootloader chain.
+      If you do not provide your own keys, debug keys are created automatically.
+      However, you should never go into production with an application that is not protected by secure keys.
+
+#. Run ``menuconfig`` on your application to enable Secure Boot:
+
+   a. Select **Project** > **Configure nRF Connect SDK project**.
+   #. Go to **Modules** > **Nordic nRF Connect** and select **Use Secure Bootloader** to enable :option:`CONFIG_SECURE_BOOT`.
+   #. Under **Private key PEM file** (:option:`CONFIG_SB_SIGNING_KEY_FILE`), enter the path to the private key that you created.
+      If you choose to run the sample with default debug keys, you can skip this step.
+
+      There are additional configuration options that you can modify, but it is not recommended to do so.
+      The default settings are suitable for most use cases.
+
+      .. note::
+         If you need more flexibility with signing, or if you do not want the build system to handle your private key, choose :option:`CONFIG_SB_SIGNING_CUSTOM`.
+         In this case, you must also specify :option:`CONFIG_SB_SIGNING_COMMAND` and :option:`CONFIG_SB_SIGNING_PUBLIC_KEY`.
+
+   #. Click **Configure**.
+
+#. Select **Build** > **Build Solution** to compile your application.
+   The build process creates two images, one for the bootloader and one for the application, and merges them together.
+#.  Select **Build** > **Build and Run** to program the resulting image to your device.
+
+Building the bootloader as stand-alone sample
+=============================================
+
+For testing purposes or if you want to program the bootloader independently of your application, you can build the bootloader sample as stand-alone sample.
+
+.. |sample path| replace:: :file:`samples/bootloader/`
+
+.. include:: /includes/build_and_run.txt
+
+Testing
+=======
+
+To test the bootloader sample, :ref:`add it to any other sample <bootloader_adding>` and build that sample as usual.
+
+After programming the sample to your board, test it by performing the following steps:
+
+#. |connect_terminal|
+#. Reset the board.
+#. Observe that the kit prints the following information::
+
+      Attempting to boot from address 0x8000.
+
+      Verifying signature against key 0.
+
+      Signature verified.
+
+      Booting (0x8000).
+
+Dependencies
+************
+
+This sample uses the following |NCS| libraries:
+
+* :ref:`partition_manager`
+* :ref:`fw_info`
+* :ref:`fprotect_readme`
+* ``include/bl_validation.h``
+* ``include/bl_crypto.h``
+* ``subsys/bootloader/include/provision.h``
+
+In addition, it uses the following Zephyr libraries:
+
+* ``include/zephyr/types.h``
```

Comment:
I don't see any value of stating this (any of the zephyr libraries). All samples use these libraries, but maybe having this here is convention for our samples.

Reply:
We have a task to define which libraries to include. I'll remove the Zephyr ones for now.